### ISSUE TITLE
beatmap download timeout

### ISSUE THREAD

**Issue by fantasyzhjk** [ppy/osu#25008]
### Type

Other

### Bug description

I encountered an issue where the osu!lazer client doesn't utilize the system's proxy settings, causing timeouts when downloading beatmaps, especially in network environments like China. Unlike osu!stable, which automatically retrieves system proxy information, osu!lazer currently does not.

### Screenshots or videos

![image](https://github.com/ppy/osu/assets/57699176/dd9cae91-4647-4581-9e9d-4d485ec9bcec)

### Version

2023.924.1

### Logs

[runtime.log](https://github.com/ppy/osu/files/12804625/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12804626/updater.log)
[database.log](https://github.com/ppy/osu/files/12804627/database.log)
[input.log](https://github.com/ppy/osu/files/12804629/input.log)
[network.log](https://github.com/ppy/osu/files/12804630/network.log)
[performance.log](https://github.com/ppy/osu/files/12804631/performance.log)

**Comment by DGCK81LNN** [comment]
i would really appreciate it if lazer had its own proxy setting, i have problems connecting to the osu server but my obs streaming should not use a proxy

**Comment by ChiNoel-osu** [comment]
I toggle my system proxy a lot, I play lazer sometimes with and sometimes without proxy. And occasionally for some reason lazer seems to hold dead on to my configured proxy setting even if I turned system proxy setting OFF (the game would just constantly reconnecting to a proxy server that doesn't exist anymore), `HTTP_PROXY` and `HTTPS_PROXY` are not set either. A system restart fixes this.
And I'm not able to reproduce it now it's so weird.

**Comment by ChiNoel-osu** [comment]
Yep, the game is not picking up system proxy setting changes.
![image](https://github.com/user-attachments/assets/f00a0274-1f5f-49c9-9cb0-e5bbe545dede)

**Comment by learningis1st** [comment]
Also can confirm on macOS
![截屏2024-06-27 上午10 40 49](https://github.com/user-attachments/assets/93060d27-91e4-4a22-a8c7-651c2a8de8b5)
![截屏2024-06-27 上午10 41 31](https://github.com/user-attachments/assets/f61aad2b-d69e-4204-99fc-c0ce8d4d9391)

**Comment by peppy** [comment]
> Yep, the game is not picking up system proxy setting changes. ![image](https://private-user-images.githubusercontent.com/39649720/363499281-f00a0274-1f5f-49c9-9cb0-e5bbe545dede.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUyNDQwOTksIm5iZiI6MTcyNTI0Mzc5OSwicGF0aCI6Ii8zOTY0OTcyMC8zNjM0OTkyODEtZjAwYTAyNzQtMWY1Zi00OWM5LTljYjAtZTViYmU1NDVkZWRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTAyVDAyMjMxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2OGVkMzhhZGEwNmU1MjBhN2M2MDZmNWNkMjYwZmJlNmZjYzkxNTZlYjQyNDg4YTgzYzc0MWVjNzk3ZjI3MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JH86Md_ZcmUgvB1PIBMyvTeAkpwAlV5Wxjrwi_KFUsI)

are you restarting the game after changing the settings?

**Comment by ChiNoel-osu** [comment]
> are you restarting the game after changing the settings?

Yes, I restarted the game (waited till osu!.exe disappears in taskmgr) after I change the proxy settings.

**Comment by ChiNoel-osu** [comment]
Had a feeling that this might have something to do with the underlying `HttpClient`.
So I did some experiments myself:

![Image](https://github.com/user-attachments/assets/973e38ac-a513-4264-87aa-cd9323d5f8f1)

And yeah somehow the default proxy is the now already disabled proxy setting.

From [.NET doc](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultproxy) it says the default setting is initialized from env or system proxy setting. Since I don't have the env set it must be something wrong with it. I even changed my proxy setting to something like 123.4.5.6 but it's still getting the old address.

Don't know if this is it but I see in [osu.Framework.IO.Network.WebRequest](https://github.com/ppy/osu-framework/blob/master/osu.Framework/IO/Network/WebRequest.cs) the `Proxy` property is not set in `SocketsHttpHandler` so it uses the default one. Maybe explicitly setting both `UseProxy` and `Proxy` there and expose them to user could help?
